src/plugins/NEMD/RegionSampling.cpp

```diff
@@ -286,13 +296,22 @@ void SampleRegion::readXML(XMLfileUnits& xmlconfig)
 				{
 					xmlconfig.changecurrentnode(nodeIter);
 					uint32_t cid = 0;
-					if(not (xmlconfig.getNodeValue("@cid", cid) && cid < _numComponents) )
+					if( (cid > _numComponents) || ((not xmlconfig.getNodeValue("@cid", cid)) && (not _boolSingleComp)) ){
```

Comment:
this looks incorrect (as LGTM) complains:
when you start there, cid = 0 and _numComponents >=0, therefore the first if is always false.
only after that evaluation cid is actually read from the xml.

Reply:
so simply moving the cid check to the right will suffice, but it is probably better to take the xml call outside of the if statement.

Reply:
Sure, I fixed it.